agent/bench-scripts/pbench-fio

```diff
@@ -801,21 +807,21 @@ function fio_run_benchmark() {
 	typeset -i ntesttypes=$(echo $test_types | sed -e 's/,/ /g' | wc -w)
 	typeset -i nblocksizes=$(echo $block_sizes | sed -e 's/,/ /g' | wc -w)
 	typeset -i total_iterations=$ntargets*$ntesttypes*$nblocksizes
+	local iteration_failures=0
 	for dev in $targets; do
 		for test_type in `echo $test_types | sed -e s/,/" "/g`; do
 			for block_size in `echo $block_sizes | sed -e s/,/" "/g`; do
-				job_num=1
-				iteration="${count}-${test_type}-${block_size}KiB"
+				local job_num=1
+				local iteration="${count}-${test_type}-${block_size}KiB"
 				if [[ "$job_mode" = "serial" ]]; then
 					dev_short_name="$(basename $dev)"
 					# easier to identify what job used what device when having 1 job per device
 					iteration="${iteration}-${dev_short_name}"
 				fi
 				record_iteration ${count} ${test_type} ${block_size} ${dev} ${iteration}
-				iteration_dir=$benchmark_run_dir/$iteration
-				result_stddevpct=$maxstddevpct # this test case will get a "do-over" if the stddev is not low enough
-				failures=0
-				while [[ $(echo "if (${result_stddevpct} >= ${maxstddevpct}) 1 else 0" | bc) -eq 1 ]]; do
```

Comment:
I don't quite see what happened to the "do-over if stddev is not low enough" logic ... but then I must have missed where `result_stddevpct` was set in the loop, because I don't see it. Somewhere I'd expect we're *not* exiting even if everything was successful if the stddev is too high? I see this same pattern repeated across the various wrappers, but nowhere under the `agent` tree do I see anything that *sets* `result_stddevpct` except to initialize it to `$maxstddevpct`. Weird.

Reply:
I think it was being initialized so that the loop would run, and then the script which should/would have updated the value is instead returning a failure status if the value is out of bounds...it was just a fancily-named control variable.  What Peter has in there now is much better, I think.

Reply:
> I don't quite see what happened to the "do-over if stddev is not low enough" logic ... but then I must have missed where `result_stddevpct` was set in the loop, because I don't see it. Somewhere I'd expect we're _not_ exiting even if everything was successful if the stddev is too high? I see this same pattern repeated across the various wrappers, but nowhere under the `agent` tree do I see anything that _sets_ `result_stddevpct` except to initialize it to `$maxstddevpct`. Weird.

I agree, this code is weird.  Ideally, the inner loop should be a separate method that returns a value to be tested for the control loop, but fixing these nested loops is a level of refactoring best left to a re-write in Python3 or shedding in favor of just using https://github.com/cloud-bulldozer/benchmark-wrapper.

For many years I have read this code where `result_stddevpct` was being initialized but never changed, and just left this weirdness alone for fear of breaking something given no bandwidth to address it.  @webbnh's comments pushed me to make the change so we can more easily verify the logic is correct.

Reply:
> I think it was being initialized so that the loop would run, and then the script which should/would have updated the value is instead returning a failure status if the value is out of bounds...it was just a fancily-named control variable. What Peter has in there now is much better, I think.

While I suspect that, on the surface, all of that is true, the point of this variable was clearly to *continue running even on success* if the standard deviation was above a threshhold; and an error exit deeper in the code can't accomplish that. Maybe there's something deeper that's looping without dropping out to this code, but I didn't find it on a quick scan. It seems likely that this code isn't doing what some developer once thought they wanted it to do; what it *is* doing (in this respect), or *how* that's happening, isn't at all clear... or whether that represents a problem we ought to consider worrying about. 😀

Reply:
> For many years I have read this code where `result_stddevpct` was being initialized but never changed, and just left this weirdness alone for fear of breaking something given no bandwidth to address it. @webbnh's comments pushed me to make the change so we can more easily verify the logic is correct.

Ha; yeah, I know that feeling. So basically you have no more insight into this bizarre piece of historical logic than I do, we know that it's not doing anything sensible (and is in addition "not doing it" in a completely bizarre way); so any problem we have after removing it, we probably already had, and we're better off for losing the cruft. 😆